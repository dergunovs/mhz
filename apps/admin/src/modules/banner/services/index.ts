import { ComputedRef, Ref } from 'vue';

import { api, useQuery, useMutation, IPageQuery, convertParams } from 'mhz-helpers';
import { API_BANNER, IBanner, IBaseReply, IFilterData } from 'mhz-contracts';

export function getBanners(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: IBanner[]; total: number; filters: IFilterData }>(API_BANNER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_BANNER, query], queryFn: fn });
}

export function getBanner(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: IBanner }>(`${API_BANNER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_BANNER, id], queryFn: fn });
}

export function postBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.post<IBaseReply>(API_BANNER, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function updateBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.patch<IBaseReply>(`${API_BANNER}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function deleteBanner(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_BANNER}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}
