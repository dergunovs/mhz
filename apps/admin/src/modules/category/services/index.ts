import { ComputedRef } from 'vue';

import { api, useQuery, useMutation } from 'mhz-helpers';
import { API_CATEGORY, ICategory, IBaseReply } from 'mhz-contracts';

export function getCategories(options?: object) {
  async function fn() {
    const { data } = await api.get<{ data: ICategory[] }>(API_CATEGORY);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY], queryFn: fn, ...options });
}

export function getCategory(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICategory }>(`${API_CATEGORY}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CATEGORY, id], queryFn: fn });
}

export function postCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.post<IBaseReply>(API_CATEGORY, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function updateCategory(options: object) {
  async function fn(formData: ICategory) {
    const { data } = await api.patch<IBaseReply>(`${API_CATEGORY}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}

export function deleteCategory(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_CATEGORY}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_CATEGORY], mutationFn: fn, ...options });
}
