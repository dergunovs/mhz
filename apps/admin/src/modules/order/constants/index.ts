export const URL_ORDER = '/orders';
