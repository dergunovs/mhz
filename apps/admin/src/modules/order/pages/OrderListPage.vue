<template>
  <div>
    <PageTitle>Заказы</PageTitle>

    <div>Заказы</div>
  </div>
</template>

<script setup lang="ts">
import PageTitle from '@/layout/components/PageTitle.vue';
</script>
