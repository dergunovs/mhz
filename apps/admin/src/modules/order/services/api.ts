import { ComputedRef, Ref } from 'vue';

import { api, IPageQuery, convertParams } from 'mhz-helpers';
import { API_ORDER, IBaseReply, IOrder, TOrderStatus } from 'mhz-contracts';

export async function getOrdersApi(query: Ref<IPageQuery | number>) {
  const params = convertParams(query);

  const { data } = await api.get<{ data: IOrder[]; total: number }>(API_ORDER, { params });

  return data;
}

export async function getOrderApi(id?: ComputedRef<string | string[]>) {
  if (!id?.value) return null;

  const { data } = await api.get<{ data: IOrder }>(`${API_ORDER}/${id.value}`);

  return data.data;
}

export async function updateOrderApi(order: { status: TOrderStatus; id?: string }) {
  if (!order.id) return null;

  const { data } = await api.patch<IBaseReply>(`${API_ORDER}/${order.id}`, { status: order.status });

  return data;
}

export async function deleteOrderApi(id?: string) {
  if (!id) return null;

  const { data } = await api.delete<IBaseReply>(`${API_ORDER}/${id}`);

  return data;
}
