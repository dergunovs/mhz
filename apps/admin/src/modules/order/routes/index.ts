import { URL_ORDER } from '@/order/constants';

export const orderRoutes = [
  {
    path: URL_ORDER,
    name: 'OrderList',
    component: () => import('@/order/pages/OrderListPage.vue'),
  },
  {
    path: `${URL_ORDER}/:order`,
    name: 'Order',
    component: () => import('@/order/pages/OrderPage.vue'),
  },
];
