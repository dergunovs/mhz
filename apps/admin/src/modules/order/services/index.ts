import { ComputedRef, Ref } from 'vue';

import { IOrder, TOrderStatus } from 'mhz-types';
import { api, useQuery, useMutation, IPageQuery } from 'mhz-helpers';

import { API_ORDER } from '@/order/constants';

export function getOrders(query: Ref<IPageQuery | number>) {
  async function fn(): Promise<{ data: IOrder[]; total: number }> {
    const params =
      typeof query.value === 'number'
        ? { page: query.value }
        : {
            page: query.value.page || 1,
            sort: query.value.sort.value,
            dir: query.value.sort.isAsc === false ? 'desc' : 'asc',
          };

    const { data } = await api.get(API_ORDER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_ORDER, query], queryFn: fn });
}

export function getOrder(id?: ComputedRef<string | string[]>) {
  async function fn(): Promise<IOrder | null> {
    if (!id?.value) return null;

    const { data } = await api.get(`${API_ORDER}/${id.value}`);

    return data;
  }

  return useQuery({ queryKey: [API_ORDER, id], queryFn: fn });
}

export function updateOrder(id: ComputedRef<string | undefined>, options: object) {
  async function fn(status: TOrderStatus) {
    await api.patch(`${API_ORDER}/${id.value}`, { status });
  }

  return useMutation({ mutationKey: [API_ORDER, id], mutationFn: fn, ...options });
}

export function deleteOrder(id: ComputedRef<string | undefined>, options: object) {
  async function fn() {
    await api.delete(`${API_ORDER}/${id.value}`);
  }

  return useMutation({ mutationKey: [API_ORDER], mutationFn: fn, ...options });
}
