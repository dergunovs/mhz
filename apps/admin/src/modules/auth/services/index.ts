import { Ref } from 'vue';
import { useMutation, useQuery } from '@tanstack/vue-query';

import { ILoginFormData } from '@/auth/interface';
import { API_GET_CHECK_AUTH, API_POST_LOGIN } from '@/auth/constants';

import { api } from '@/common/services/api';

export function postLogin(formData: Ref<ILoginFormData>, options: object) {
  async function fn(): Promise<string | undefined> {
    const { data } = await api.post(API_POST_LOGIN, formData.value);

    return data.token;
  }

  return useMutation({
    mutationKey: [API_POST_LOGIN, formData],
    mutationFn: fn,
    ...options,
  });
}

export function getCheckAuth(options: object) {
  async function fn() {
    const { data } = await api.get(API_GET_CHECK_AUTH);

    return data;
  }

  return useQuery({
    queryKey: [API_GET_CHECK_AUTH],
    queryFn: fn,
    ...options,
  });
}
