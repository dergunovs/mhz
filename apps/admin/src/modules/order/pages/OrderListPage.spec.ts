import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { IOrder } from 'mhz-contracts';
import { dataTest } from 'mhz-helpers';

import OrderListPage from './OrderListPage.vue';

import { mockQueryReply, wrapperFactory } from '@/common/test';
import * as orderServices from '@/order/services';
import { ORDERS } from '@/order/fixtures';

const spyGetOrders = vi.spyOn(orderServices, 'getOrders').mockReturnValue(mockQueryReply(ORDERS));

const orderList = dataTest('order-list-page-list');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(OrderListPage, {});
});

enableAutoUnmount(afterEach);

describe('OrderListPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(OrderListPage)).toBeTruthy();
  });

  it('gets orders and passes them to list props', async () => {
    expect(spyGetOrders).toBeCalledTimes(1);

    expect(wrapper.findComponent<DefineComponent<{ orders: IOrder[] }>>(orderList).vm.$props.orders).toEqual(
      ORDERS.data
    );
  });
});
