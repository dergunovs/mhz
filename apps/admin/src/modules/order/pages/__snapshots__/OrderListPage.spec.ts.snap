// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderListPage > matches snapshot 1`] = `
"<div>
  <page-title-stub links="[object Object],[object Object]"></page-title-stub>
  <div class="_page_259553">
    <order-list-stub orders="[object Object],[object Object]" modelvalue="[object Object]" data-test="order-list-page-list"></order-list-stub>
    <ui-pagination-stub page="1" total="1" lang="en"></ui-pagination-stub>
  </div>
</div>"
`;
